src/policy/offload.rs

```diff
@@ -0,0 +1,249 @@
+
+pub struct OffloadPolicy {}
+
+impl PolicyModule for OffloadPolicy {
+    fn init() -> Self {
+        OffloadPolicy {}
+    }
+
+    fn name() -> &'static str {
+        "Offload Policy"
+    }
+
+    fn trap_from_payload(
+        &mut self,
+        mctx: &mut MiralisContext,
+        ctx: &mut VirtContext,
+    ) -> PolicyHookResult {
+        let trap_info = ctx.trap_info.clone();
+
+        match trap_info.get_cause() {
+            MCause::LoadAddrMisaligned => self.emulate_misaligned_read(ctx, mctx),
+            MCause::StoreAddrMisaligned => self.emulate_misaligned_write(ctx, mctx),
+            MCause::EcallFromSMode => {
+                let timer_eid: bool = ctx.get(Register::X17) == sbi_codes::SBI_TIMER_EID;
+                let timer_fid: bool = ctx.get(Register::X16) == sbi_codes::SBI_TIMER_FID;
+
+                if timer_eid && timer_fid {
+                    let v_clint = Plat::get_vclint();
+                    v_clint.write_clint_from_payload(ctx, mctx, ctx.regs[Register::X10 as usize]);
+                    ctx.pc += 4;
+                    PolicyHookResult::Overwrite
+                } else {
+                    PolicyHookResult::Ignore
+                }
+            }
+            MCause::IllegalInstr => {
+                let instr = unsafe { get_raw_faulting_instr(&ctx.trap_info) };
+
+                let is_privileged_op: bool = instr & 0x7f == 0b111_0011;
+                let is_time_register: bool = (instr >> 20) == 0b1100_0000_0001;
+
+                if is_privileged_op && is_time_register {
+                    let rd = (instr >> 7) & 0b11111;
+                    let _rs1 = (instr >> 15) & 0b11111;
+
+                    let func3_mask = instr & 0b111000000000000;
+                    match func3_mask {
+                        0x2000 => {
+                            ctx.set(Register::try_from(rd).unwrap(), Arch::read_csr(Csr::Time));
+                            ctx.pc += 4;
+                            return PolicyHookResult::Overwrite;
+                        }
+                        0x1000 | 0x3000 | 0x5000 | 0x6000 | 0x7000 => {
+                            todo!("Handle the offload of other CSR instructions")
+                        }
+                        _ => {}
+                    }
+                }
+
+                PolicyHookResult::Ignore
+            }
+            _ => PolicyHookResult::Ignore,
+        }
+    }
+
+    fn ecall_from_payload(
+        &mut self,
+        _mctx: &mut MiralisContext,
+        _ctx: &mut VirtContext,
+    ) -> PolicyHookResult {
+        PolicyHookResult::Ignore
+    }
+
+    fn switch_from_payload_to_firmware(
+        &mut self,
+        _ctx: &mut VirtContext,
+        _mctx: &mut MiralisContext,
+    ) {
+    }
+
+    fn switch_from_firmware_to_payload(
+        &mut self,
+        _ctx: &mut VirtContext,
+        _mctx: &mut MiralisContext,
+    ) {
+    }
+
+    fn on_interrupt(&mut self, _ctx: &mut VirtContext, _mctx: &mut MiralisContext) {}
```

Comment:
Those could be default implementations, to remove the need for empty implementations in policy modules.